planetiler-core/src/main/java/com/onthegomap/planetiler/pmtiles/ReadablePmtiles.java

```diff
@@ -0,0 +1,163 @@
+package com.onthegomap.planetiler.pmtiles;
+
+import com.onthegomap.planetiler.archive.ReadableTileArchive;
+import com.onthegomap.planetiler.geo.TileCoord;
+import com.onthegomap.planetiler.util.CloseableIterator;
+import com.onthegomap.planetiler.util.Gzip;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SeekableByteChannel;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class ReadablePmtiles implements ReadableTileArchive {
+  private SeekableByteChannel channel;
+
+  public ReadablePmtiles(SeekableByteChannel channel) {
+    this.channel = channel;
+  }
+
+  /**
+   * Finds the relevant entry for a tileId in a list of entries.
+   * <p>
+   * If there is an exact match for tileId, return that. Else if the tileId matches an entry's tileId + runLength,
+   * return that. Else if the preceding entry is a directory (runLength = 0), return that. Else return null.
+   */
+  public static WriteablePmtiles.Entry findTile(List<WriteablePmtiles.Entry> entries, long tileId) {
```

Comment:
I prefer `List<Entry>` because it makes it clear that a Directory holds no other extra information apart from its entry contents. Refactoring `WriteablePmtiles.Entry` to `Pmtiles.Entry` makes it a little less verbose.

Reply:
Ok sounds fine - it might just be nice to have the utilities for working with/parsing/serializing directories collected to a common place.